framework/aster-frame/src/boot/mod.rs

```diff
@@ -119,13 +119,26 @@ pub fn call_aster_main() -> ! {
     }
     #[cfg(ktest)]
     unsafe {
+        use alloc::boxed::Box;
+
+        use crate::task::{set_scheduler, FifoScheduler, Scheduler, TaskOptions};
+
         crate::init();
         // The whitelists that will be generated by OSDK runner as static consts.
         extern "Rust" {
             static KTEST_TEST_WHITELIST: Option<&'static [&'static str]>;
             static KTEST_CRATE_WHITELIST: Option<&'static [&'static str]>;
         }
-        run_ktests(KTEST_TEST_WHITELIST, KTEST_CRATE_WHITELIST);
+        // Set the global scheduler a FIFO scheduler.
+        let simple_scheduler = Box::new(FifoScheduler::new());
+        let static_scheduler: &'static dyn Scheduler = Box::leak(simple_scheduler);
+        set_scheduler(static_scheduler);
+
+        let test_task = move || {
+            run_ktests(KTEST_TEST_WHITELIST, KTEST_CRATE_WHITELIST);
+        };
+        let _ = TaskOptions::new(test_task).data(Box::new(1)).spawn();
+        loop {}
```

Comment:
Here the loop seems very weird to me. The exiting happens in a kernel task `run_ktests` and the main task just spawns the task and spins. I think it's better to spawn tasks in the `run_ktests` function, which won't return.

Reply:
This function `call_aster_main` need `-> ! ` as return value, so we have to add `loop {}` to match types, although it won't execute.

Reply:
Since the loop will never be reached, how about using `unreachable!()` instead of `loop{}` ?

Reply:
> Since the loop will never be reached, how about using `unreachable!()` instead of `loop{}` ?

I am confused. I think `TaskOption::spawn` will immediately return after the task is started, won't it? So it is _reachable_.

Reply:
> I am confused. I think TaskOption::spawn will immediately return after the task is started, won't it? So it is reachable.

If `TaskOption::spawn` is executed in task context, this method will return sooner or later, so the following code is **reachable**.

However, if `TaskOption::spawn` is executed in boot context (the context before spawning the first task), the following code after `spawn` in boot context will never be executed again. The code will jump to the task entry point and never jump back to boot context.

Reply:
Solved.